File: test/fixtures/typescript/extractFunction.output.ts
function processItems(items: string[]) {
  const filtered = items.filter(item => item.length > 3);
  const uppercased = filtered.map(item => item.toUpperCase());
  const result = uppercased.join(', ');
  return result;
}

function processData(items: string[]) {
  console.log('Starting to process data...');

  // Code block to extract (lines 4-7)
  const result = processItems(items);
  console.log(result);

  console.log('Processing complete');
  return items.length;
}

export { processData };

File: test/fixtures/python/extract_function.output.py
def process_items(items):
    filtered = [item for item in items if len(item) > 3]
    uppercased = [item.upper() for item in filtered]
    result = ', '.join(uppercased)
    return result

def process_data(items):
    print('Starting to process data...')
    
    # Code block to extract (lines 4-7)
    result = process_items(items)
    print(result)
    
    print('Processing complete')
    return len(items)